### PR TITLE
add unit test for kafka topic

### PR DESCRIPTION
**Pull request by mgold1234**
Signed-off-by: mgold1234 <mgold@redhat.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type EdgeConfig struct {
 	UploadWorkers            int                       `json:"upload_workers,omitempty"`
 	KafkaConfig              *clowder.KafkaConfig      `json:"kafka,omitempty"`
 	KafkaBrokers             []clowder.BrokerConfig    `json:"kafka_brokers,omitempty"`
-	KafkaTopics              map[string]string         `json:"kafka_topics,omitempty"`
+	KafkaTopics              []clowder.TopicConfig     `json:"kafka_topics,omitempty"`
```
Comment (ldjebran) [inline config/config.go:48]: do we need this config at all ? 
where it's used  and for which purpose ?

Comment (mgold1234) [reply config/config.go:48]: @ldjebran I use this config at unit test

Comment (mgold1234) [reply config/config.go:48]: @ldjebran what do you suggest me to use instead of kafkaTopic?

Comment (ldjebran) [reply config/config.go:48]: maybe KafkaConfig https://github.com/RedHatInsights/edge-api/blob/main/config/config.go#L46
That is used in the application to search for the topic and raise that error when topic not found GetTopic 
https://github.com/RedHatInsights/edge-api/blob/main/pkg/common/kafka/topics.go#L48

File: pkg/common/kafka/topics.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ const (
 type TopicNotFoundError struct{}
 
 func (e *TopicNotFoundError) Error() string {
-	return "Topic is not found in config"
+	return "topic is not found in config"
```
[the remark this answers is not in the document]
Comment (mgold1234) [reply pkg/common/kafka/topics.go]: I changed it to 'topic' because when defining errors.New , the string inside should start with small letter

**Review by tomershinhar — APPROVED** [review]
LGTM

File: pkg/common/kafka/topics_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package kafkacommon_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/magiconair/properties/assert"
```
Comment (ldjebran) [inline pkg/common/kafka/topics_test.go:8]:
```go
"github.com/stretchr/testify/assert"
```

File: pkg/common/kafka/topics_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+			ExpectedError:   errors.New("topic is not found in config"),
+		},
+	}
+	for _, test := range cases {
+		t.Run(test.Name, func(t *testing.T) {
+			cfg.KafkaConfig = test.conf
+			topic, error := service.GetTopic(topicConfig.RequestedName)
+			assert.Equal(t, topic, test.ExpectedRequest)
+			if test.ExpectedError != nil {
+				assert.Equal(t, error.Error(), test.ExpectedError.Error())
```
Comment (ldjebran) [inline pkg/common/kafka/topics_test.go:62]: need to add one assertion before line 62 
```go
if test.ExpectedError != nil {
    assert.Error(t, error, "expected error but no error occurred")
    assert.Equal(t, error.Error(), test.ExpectedError.Error())
} else {
    ...
			
```

**Review by ldjebran — APPROVED** [review]
/lgtm

**Comment by ldjebran** [comment]
@Mergifyio refresh

**Comment by ldjebran** [comment]
@Mergifyio rebase

**Pull request merged**